### PR TITLE
TLS hardening

### PR DESCRIPTION
**Pull request by bcebere**
### 1. Why is this pull request needed and what does it do?
The TLS plugin overwrites the tls.Config from Caddy, and zeroes the minimum version, maximum version and default ciphersuites.

This results in some issues when using CoreDNS with TLS support:

SSLv3, TLSv1.0 and TLSv1.1 are accepted.
The are some weak ciphersuites accepted.
No server preferred ciphersuite order is provided.
No TLS 1.3 support

### 2. Which issues (if any) are related?
https://github.com/coredns/coredns/issues/2935

### 4. Does this introduce a backward incompatible change or deprecation?
Drops support for SSLv3, TLS v1.0 and TLS v1.1

### CODE REVIEW

File: plugin/tls/tls.go [hunk 1]
```diff
@@ -25,6 +26,24 @@ func setup(c *caddy.Controller) error {
 	return nil
 }
 
+func hardenTLS(tls *ctls.Config) {
+	os.Setenv("GODEBUG", "tls13=1")
```
Comment (miekg) [inline plugin/tls/tls.go]: This looks odd and brittle, it also overwrites GODEBUG which is something we don't want. Please remove this line.

File: plugin/tls/tls.go [hunk 2]
```diff
@@ -25,6 +26,24 @@ func setup(c *caddy.Controller) error {
 	return nil
 }
 
+func hardenTLS(tls *ctls.Config) {
```
Comment (miekg) [inline plugin/tls/tls.go]: It's not really harden, it's settting some defaults that we think are good.

setTLSParameters, or just setTLSDefaults?

File: plugin/tls/tls.go (hunk 1 above)
Comment (bcebere) [reply plugin/tls/tls.go]: Unfortunately, this seems the only way to activate TLS 1.3 in Go right now.

I don't know how to do it cleaner.

Comment (bcebere) [reply plugin/tls/tls.go]: I could change it to
`os.Setenv("GODEBUG", os.Getenv("GODEBUG")+",tls13=1")`

At least I won't overwrite the variable.

I took the info from here https://github.com/golang/go/issues/30055

**Review by miekg — APPROVED** [review]

File: plugin/tls/tls.go (hunk 1 above)
Comment (bcebere) [reply plugin/tls/tls.go]: Should we wait for the default TLS 1.3 support in Go fow now? 
I am not sure why they chose the opt-in via GODEBUG

**Comment by miekg** [comment]
let's try again

/merge

**Pull request merged**